main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"os"
+	"github.com/codegangsta/cli"
+	"fmt"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "fetch"
+	app.Usage = "download a file or folder from a specific release of a GitHub repo subject to the Semantic Versioning constraints you impose"
+	app.Version = getVersion(Version, VersionPrerelease)
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name: "repo",
+			Usage: "Required. Fully qualified URL of the GitHub repo.",
+		},
+		cli.StringFlag{
+			Name: "tag",
+			Usage: "The specific git tag to download, expressed with Version Constraint Operators.\n\tIf left blank, fetch will download the latest git tag.\n\tSee https://github.com/gruntwork-io/fetch#version-constraint-operators for examples.",
+		},
+		cli.StringFlag{
+			Name: "github-oauth-token",
+			Usage: "A GitHub Personal Access Token (https://help.github.com/articles/creating-an-access-token-for-command-line-use/)\n\tThis is used to authenticate fetch to private GitHub repos.",
+		},
+	}
+
+	app.Action = func(c *cli.Context) {
+
+		repoUrl := c.String("repo")
+		tagConstraint := c.String("tag")
+		githubToken := c.String("github-oauth-token")
+
+		// TODO: process repoFilePath and localFileDst args from command line
+		repoFilePath := "/"
+		localFileDst := "/Users/josh/temp"
+
+		// Validate required args
+		if repoUrl == "" {
+			fmt.Fprintf(os.Stderr, "ERROR: The --repo argument is required. Run \"%s --help\" for full usage info.", app.Name)
+			os.Exit(1)
+		}
+
+		// Get the tags for the given repo
+		tags, err := FetchTags(repoUrl, githubToken)
+		if err != nil {
+			if err.errorCode == 401 {
+				fmt.Fprintf(os.Stderr, getErrorMessage(401, err.details))
+				os.Exit(1)
+			} else if err.errorCode == 404 {
+				fmt.Fprintf(os.Stderr, getErrorMessage(404, err.details))
+				os.Exit(1)
+			} else {
+				panic(err)
+			}
+		}
```

Comment:
> I'd recommend defining 401, 404, and other magic numbers as named constants.

Great idea. Made the code much nicer.

> Also, why are you treating those special and doing an os.Exit for them, and a panic for all other errors?

I also wasn't sure about the best way to handle this, but my reasoning is that if it's a "checked" exception, then I can print the pretty errors and exit, but if it's an unchecked exception, what else can I do but panic and die? Do you think it's worth adding pretty, descriptive messages for each of the panics instead?